src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -295,31 +297,32 @@ rpmostreed_deployment_generate_variant (OstreeSysroot *sysroot,
   { g_autoptr(GVariant) base_meta = g_variant_get_child_value (commit, 0);
     g_variant_dict_insert (&dict, "base-commit-meta", "@a{sv}", base_meta);
   }
+  variant_add_commit_details (&dict, NULL, commit);
 
   gboolean gpg_enabled = FALSE;
   g_autoptr(GVariant) sigs = NULL;
-  if (!rpmostreed_deployment_gpg_results (repo, refspec, base_checksum, &sigs, &gpg_enabled, error))
-    return NULL;
-  variant_add_commit_details (&dict, NULL, commit);
-
-  g_autofree char *pending_base_commitrev = NULL;
-  if (!ostree_repo_resolve_rev (repo, refspec, TRUE,
-                                &pending_base_commitrev, error))
-    return NULL;
-
-  if (pending_base_commitrev && strcmp (pending_base_commitrev, base_checksum) != 0)
+  if (refspec && refspec_type == RPMOSTREE_REFSPEC_TYPE_OSTREE)
```

Comment:
Why gate on `refspec` here? We should always have one, right? In fact, the origin errors out right now if there isn't one.

Similarly down below where we insert the `origin` key.